policy/modules/services/kubernetes.te

```diff
@@ -0,0 +1,534 @@
+policy_module(kubernetes)
+
+########################################
+#
+# Declarations
+#
+
+# common attribute for all kubernetes domains
+attribute kubernetes_domain;
+
+attribute_role kubernetes_roles;
+roleattribute system_r kubernetes_roles;
+
+# common attribute for all container engines
+# that may be used with kubernetes
+attribute kubernetes_container_engine_domain;
+
+# common attribute for all container domains
+# that may be used with kubernetes
+attribute kubernetes_container_domain;
+
+# attribute for all objects that may be mounted
+# on by kubernetes containers
+attribute kubernetes_mountpoint_type;
+
+# common attribute for all kubectl domains
+attribute kubectl_domain;
+
+type kubelet_t, kubernetes_domain;
+type kubelet_exec_t;
+domain_type(kubelet_t)
+container_engine_executable_file(kubelet_exec_t)
+init_daemon_domain(kubelet_t, kubelet_exec_t)
+role kubernetes_roles types kubelet_t;
+
+type kubeadm_t;
+type kubeadm_exec_t;
+application_domain(kubeadm_t, kubeadm_exec_t)
+role kubernetes_roles types kubeadm_t;
+
+type kubectl_t, kubectl_domain;
+type kubectl_exec_t;
+application_domain(kubectl_t, kubectl_exec_t)
+
+type kubernetes_plugin_t;
+corecmd_executable_file(kubernetes_plugin_t)
+kubernetes_mountpoint(kubernetes_plugin_t)
+
+type kubernetes_config_t;
+files_config_file(kubernetes_config_t)
+kubernetes_mountpoint(kubernetes_config_t)
+
+type kubernetes_runtime_t;
+files_runtime_file(kubernetes_runtime_t)
+kubernetes_mountpoint(kubernetes_runtime_t)
+
+# files created in /tmp by kubectl for editing
+type kubernetes_tmp_t;
+files_tmp_file(kubernetes_tmp_t)
+
+type kubernetes_tmpfs_t;
+files_type(kubernetes_tmpfs_t)
+kubernetes_mountpoint(kubernetes_tmpfs_t)
+
+type kubernetes_unit_t;
+init_unit_file(kubernetes_unit_t)
+
+type kubernetes_conf_home_t;
+xdg_config_content(kubernetes_conf_home_t)
+
+########################################
+#
+# common kubernetes container engine policy
+#
+
+allow kubernetes_container_engine_domain kubernetes_mountpoint_type:dir search_dir_perms;
+allow kubernetes_container_engine_domain kubernetes_mountpoint_type:dir_file_class_set { getattr mounton };
+
+ps_process_pattern(kubernetes_container_engine_domain, kubernetes_container_domain)
+
+# for some storage class providers
+dev_add_entry_generic_dirs(kubernetes_container_engine_domain)
+dev_manage_generic_files(kubernetes_container_engine_domain)
```

Comment:
I think I may have asked this before, but can we avoid `device_t` files?

Reply:
I'll dig into this further and see if we can do that.

Reply:
This is needed because of `/dev/termination-log` which is the default path for pods' `terminationLogPath`. Users can set this path to a different location to avoid writing in a container's `/dev`, but I think we can also work around this with a `filetrans()` to a private type for generic container devices.

Reply:
I think we can actually just have these dev nodes be `container_file_t` since all the rules are already in place for that. I don't see any immediate value in a separate type for this.

Reply:
Actually, no, since that would automatically associate `container_file_t` with device filesystems which is probably not what we want.